how to deploy (#10760)
zkSNACKs/WalletWasabi: WalletWasabi.Documentation/HowToDeploy.md

# The agreed procedure to deploy something to production

> 1. Create a summary of what is going to be deployed (we are still working on this and don't have anything automatic yet).

Author: For the last deployment to testnet I used a label `affiliate` so Satoshi Labs and Kukks can see what is are the changes that they could be more interested on. This is not perfect but let see how it works and in case it is not enough we can try something else or more elaborated.

> 1. Create a summary of what is going to be deployed (we are still working on this and don't have anything automatic yet).

Reviewer: If they are OK with that it is perfect for me. Can you add that to the document?

2. Announce in the Slack Integrations channel the deploy one specific commit to testnet using the @aqui tag and share the summary.
3. Affiliates must acknowledge the notification.
4. Affiliates must express whether they are willing to test what has been deployed or not.

>    1. In case Affiliates **are not** interested in testing it then the Wasabi team will notify the planned deploy date to production, and that's it.

Reviewer 2: ```suggestion
   1. In case Affiliates **are not** interested in testing it, then the Wasabi team will notify the planned deployment date to production, and that's it.
```

>    2. In case Affiliates **are** interested in testing it they must:

Reviewer 2: ```suggestion
   2. In case Affiliates **are** interested in testing it, they must:
```

      1. Let the Wasabi team know how much time they need to test it.

>       2. Share their findings. In case there is any then iterate.

Reviewer 2: ```suggestion
      2. Share their findings. In case there is any, then iterate.
```

      3. Give the Wasabi team the final approval.
      4. The Wasabi team and the Affiliates must agree on a release date for production deployment.

> # How the deploy

Reviewer 2: ```suggestion
# How to deploy
```

> 1. Create a tag that identifies the commit to be deployed. The naming format is `version-codename-iteration`, where `version` being  the official version number (for example, v2.0.3.1), `codename` an identifier of what is being deployed (for example, `wgpt`, `wabisabi`, `affsvr` or any other) and `iteration` a sequence number to identify a sequence of deployments (first iteration doesn't need to have this number). For example:

Reviewer 2: ```suggestion
1. Create a tag that identifies the commit to be deployed. The naming format is `version-codename-iteration`, where `version` is the official version number (for example, v2.0.3.1), `codename` an identifier of what is being deployed (for example, `wgpt`, `wabisabi`, `affsvr` or any other), and `iteration` a sequence number to identify a sequence of deployments (first iteration doesn't need to have this number). For example:
```

   - `v2.0.3-wabisabi` is the tag that was used to deploy the backend with the external WabiSabi library.
   - `v2.0.2.1-as` is the tag that was used to deploy the backend containing the **a**filiation **s**erver integration.
   - `v2.0.2.1-as-1` is the tag that was used to deploy the affiliation server integration again.
   - `v2.0.2.1-as-2` is the tag that was used to deploy the affiliation server integration once more.

>    - `v2.0.3wgpt` (now it should be named as `v2.0.3-wgpt`) is the tag that was used to deploy the version containing the gpt compatible version.

Reviewer 2: ```suggestion
   - `v2.0.3wgpt` (now it should be named as `v2.0.3-wgpt`) is the tag that was used to deploy the version containing the gpt-compatible version.
```

>    Note: As you can see, the deployer can chose the `codename` freely.

Reviewer 2: ```suggestion
   Note: As you can see, the deployer can choose the `codename` freely.
```

# The agreed procedure to deploy something to production

> 1. Create a summary of what is going to be deploy (we are still working on this and don't have anything automatic yet)

Reviewer 3: ```suggestion
1. Create a summary of what is going to be deployed (we are still working on this and don't have anything automatic yet)
```

2. Announce in the Slack Integrations channel the deploy one specific commit to testnet using the @aqui tag and share the summary
3. Affiliates must acknowledge the notification

> 4. Affiliates must express whether they are willing to test what had been deployed or not

Reviewer 3: ```suggestion
4. Affiliates must express whether they are willing to test what has been deployed or not
```

> 4.1. In case Affiliates **are not** interested in testing it then Wasabi team will notify the planned deploy date to productioned and that's it

Reviewer 3: ```suggestion
4.1. In case Affiliates **are not** interested in testing it then the Wasabi team will notify the planned production deploy date, and that's it
```

4.2. In case Affiliates **are** interested in testing it they must:

> 4.2.1. Let the Wasabi team to know how much time they need to test it.

Reviewer 3: ```suggestion
4.2.1. Let the Wasabi team know how much time they need to test it.
```

4.2.2. Share their findings. In case there is any then iterate

> 4.4.3. Give the Wasabi team the final okay.

Reviewer 3: ```suggestion
4.4.3. Give the Wasabi team the final approval.
```

> 4.4.4. The Wasabi team and the affiliate must agree a relase date to production

Reviewer 3: ```suggestion
4.4.4. The Wasabi team and the affiliate(s) must agree a release date for production deployment
```

# How the deploy

> 1. Create a tag that identify the commit to me deployed. The naming format is `version-codename-iteration` being `version` the official version number (for example, v2.0.3.1), `codename` a identifier of what is being deployed (for example, `wgpt`, `wabisabi`, `affsvr` or any other) and `iteration` a sequence number to identify a sequence of deployments (first iteration doesn't need to have this number). For example:

Reviewer 3: ```suggestion
1. Create a tag that identifies the commit to be deployed. The naming format is `version-codename-iteration`, where`version` being  the official version number (for example, v2.0.3.1), `codename` a identifier of what is being deployed (for example, `wgpt`, `wabisabi`, `affsvr` or any other) and `iteration` a sequence number to identify a sequence of deployments (first iteration doesn't need to have this number). For example:
```

* `v2.0.3-wabisabi` is the tag that was used to deploy the backend with the external WabiSabi library.
* `v2.0.2.1-as` is the tag that was used to deploy the backend containing the **a**filiation **s**erver integration
* `v2.0.2.1-as-1` is the tag that was used to deploy the affiliation server integration again 
* `v2.0.2.1-as-2` is the tag that was used to deploy the affiliation server integration once more 
* `v2.0.3wgpt` (now it should be named as `v2.0.3-wgpt`) is the tag that was used to deploy the version containing the gpt compatible version

> Note: as you can see the deployer can chose the `codename` freely.

Reviewer 3: ```suggestion
Note: as you can see, the deployer can chose the `codename` freely.
```
